pkg/engine/health/health.go

```diff
@@ -27,6 +27,7 @@ func IsHealthy(obj runtime.Object) error {
 	}
 
 	objUnstructured := &unstructured.Unstructured{Object: unstructMap}
+	objUnstructured.SetGroupVersionKind(obj.GetObjectKind().GroupVersionKind())
```

Comment:
🤷‍♀ otherwise GVK is not propagated and we need that for health

Reply:
^^ this should be a comment in code ;)